### PR TITLE
KYLIN-4613 add build cubev2 and JobRestClient

### PR DESCRIPTION
**Pull request by bigxiaochu**

### CODE REVIEW

**Comment by RupengWang** [comment]
+ 1
Test Passed with the following command:
```shell
./bin/kylin.sh org.apache.kylin.tool.job.CubeBuildingCLIV2 --cube kylin_sales_cube --endTime 1338508800000 --host localhost --port 7971 --userName ADMIN --password KYLIN
```

**Comment by bigxiaochu** [comment]
@xiaowangyu how do you see.

**Comment by xiaowangyu** [comment]
> @xiaowangyu how do you see.

I think the change of JobRestClient and CubeBuildingCLIV2 can be update in original class may be better. Compatible with the original implementation. For users, the interface is unified.

**Comment by bigxiaochu** [comment]
@xiaowangyu  do you mean change of CubeBuilding? 
It will be umcompatibility . I think compatibility is import, so use v2

File: core-common/src/main/java/org/apache/kylin/common/KylinConfig.java [hunk 1]
```diff
@@ -134,7 +134,7 @@ private static void buildDefaultOrderedProperties() {
         }
     }
 
-    public static KylinConfig getInstanceFromEnv() {
+    public static KylinConfig getInstanceFromEnv(boolean allowConfigFileNoExist) {
```
Comment (hit-lacus) [inline core-common/src/main/java/org/apache/kylin/common/KylinConfig.java:137]: Why we need `allowConfigFileNoExist` ?

Comment (bigxiaochu) [reply core-common/src/main/java/org/apache/kylin/common/KylinConfig.java:137]: 因为作为客户端提交建模、构建请求时，可以不需要有kylin的部署结构，也不需要有符合部署目录结构规范的配置文件。以减少客户端程序的依赖复杂度。

**Review by hit-lacus — APPROVED** [review]
Approved

**Comment by hit-lacus** [comment]
This patch looks good to me, thanks @bigxiaochu . We should consider to write documentation for this tools.

**Pull request merged**